### PR TITLE
Enable location sharing

### PR DESCRIPTION
**Pull request by andybalaam**
---
Here's what your changelog entry will look like:

## ✨ Features
 * Enable location sharing ([\#7703](https://github.com/matrix-org/matrix-react-sdk/pull/7703)). Contributed by @andybalaam.

Preview: https://61fa64b43d0953fd0bc6bccb--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by t3chguy** [comment]
Needs doc removal from labs.md

File: src/components/views/messages/MessageEvent.tsx [hunk 1]
```diff
@@ -125,19 +125,17 @@ export default class MessageEvent extends React.Component<IProps> implements IMe
                 BodyType = UnknownBody;
             }
 
-            // TODO: this can be done in eventTypes when Polls stabilise
+            // TODO: this can be done in eventTypes when Polls stabilises
```
Comment (t3chguy) [inline src/components/views/messages/MessageEvent.tsx:128]: this comment should be updated to specify that it isn't the feature stabilising (which is happening in this PR), but the MSC/spec presumably?

File: src/components/views/messages/MessageEvent.tsx [hunk 2]
```diff
@@ -125,19 +125,17 @@ export default class MessageEvent extends React.Component<IProps> implements IMe
...
             if (M_POLL_START.matches(type)) {
                 BodyType = sdk.getComponent('messages.MPollBody');
             }
 
+            // TODO: this can be done in eventTypes when location sharing stabilises
```
Comment (t3chguy) [inline src/components/views/messages/MessageEvent.tsx:133]: same as above

File: src/settings/Settings.tsx [hunk 3]
```diff
@@ -394,12 +387,6 @@ export const SETTINGS: {[setting: string]: ISetting} = {
         default: true,
         controller: new UIFeatureController(UIFeature.Widgets, false),
     },
-    "MessageComposerInput.showLocationButton": {
-        supportedLevels: LEVELS_ACCOUNT_SETTINGS,
-        displayName: _td('Enable location sharing'),
-        default: true,
-        controller: new IncompatibleController("feature_location_share", false, false),
-    },
```
Comment (t3chguy) [inline src/settings/Settings.tsx:402]: Any reason this setting is being removed? Stickers have a similar setting

**Review by t3chguy — APPROVED** [review]
LGTM on the surface, but @matrix-org/product it'd be good to just double check that you're happy for this feature to be entirely missing on the platform we're pushing users to for improved security & message search (Desktop) - if we re-enabled manual location sharing at least it'd be mostly functional on Desktop

File: src/settings/Settings.tsx (hunk 3 above)
Comment (andybalaam) [reply src/settings/Settings.tsx:402]: More settings === bad?  The guide from Product was to remove it completely.  It was originally added to help privacy-conscious users, but there was push-back from the community that it was pointless.

Comment (t3chguy) [reply src/settings/Settings.tsx:402]: Sure, as long as it was a request by product then all good! Now with most of the composer actions living in an overflow menu it is much less of a concern for screen real estate anyway

File: src/components/views/messages/MessageEvent.tsx (hunk 1 above)
Comment (andybalaam) [reply src/components/views/messages/MessageEvent.tsx]: Fixed in e11106d1e8

File: src/components/views/messages/MessageEvent.tsx (hunk 2 above)
Comment (andybalaam) [reply src/components/views/messages/MessageEvent.tsx]: Fixed in e11106d1e8

**Comment by andybalaam** [comment]
> Needs doc removal from labs.md

https://github.com/vector-im/element-web/pull/20863

**Comment by andybalaam** [comment]
> LGTM on the surface, but @matrix-org/product it'd be good to just double check that you're happy for this feature to be entirely missing on the platform we're pushing users to for improved security & message search (Desktop) - if we re-enabled manual location sharing at least it'd be mostly functional on Desktop

Reply from Jake: it's not ideal, but we are going with it for now.

**Pull request merged**